diff --git a/addressfield.module b/addressfield.module
index 7ea844e..fd6b529 100644
--- a/addressfield.module
+++ b/addressfield.module
@@ -118,6 +118,17 @@ function addressfield_generate($address, array $handlers, array $context = array
     if ($callback = ctools_plugin_load_function('addressfield', 'format', $handler, 'format callback')) {
       $callback($format, $address, $context);
     }
+    else {
+      $plugin_types = ctools_plugin_get_plugin_type_info();
+
+      foreach($plugin_types as $module => $plugin_type) {
+        if(key_exists('format', $plugin_type)) {
+          if ($callback = ctools_plugin_load_function($module, 'format', $handler, 'format callback')) {
+            $callback($format, $address, $context);
+          }
+        }
+      }
+    }
   }

   // Store the address in the format, for processing.
